RoboJackets/robocup-software
# Added unassigned_role and tactic role request edge case

## Description
Added an unassigned_role and logic to handle unassigned robots and tactics which have too many roles relative to available robots.
**FOR FUTURE TACTIC WRITERS:** After creating new role requests, wait until len(self.assigned_roles) == # of needed roles to continue through your tactic!

## Steps to test
1. Run `ros2 launch rj_robocup sim.launch.py`with a play and tactics that will have robots unassigned
2. Run the play in the simulator

Expected result: In the command prompt, the message "UnassignedRole created for: {robot info}" should be printed as many times as there are unassigned robots
